src/ITensorsExtensions/ITensorsExtensions.jl

```diff
@@ -52,16 +54,26 @@ invsqrt_diag(it::ITensor) = map_diag(inv ∘ sqrt, it)
 pinv_diag(it::ITensor) = map_diag(pinv, it)
 pinvsqrt_diag(it::ITensor) = map_diag(pinv ∘ sqrt, it)
 
-function map_itensor(
-  f::Function, A::ITensor, lind=first(inds(A)); regularization=nothing, kwargs...
+function map_eigenvalues(
+  f::Function, A::ITensor, linds=Index[first(inds(A))]; ishermitian=false, kwargs...
 )
-  USV = svd(A, lind; kwargs...)
-  U, S, V, spec, u, v = USV
-  S = map_diag(s -> f(s + regularization), S)
-  sqrtDL, δᵤᵥ, sqrtDR = sqrt_decomp(S, u, v)
-  sqrtDR = denseblocks(sqrtDR) * denseblocks(δᵤᵥ)
-  L, R = U * sqrtDL, V * sqrtDR
-  return L * R
+  if isdiag(A)
+    return map_diag(s -> f(s), A)
+  end
+
+  @assert ishermitian
+  rinds = setdiff(inds(A), linds)
+
+  D, U = eigen(A, linds, rinds; ishermitian, kwargs...)
+  ul, ur = noncommonind(D, U), commonind(D, U)
+  ulnew = sim(ul)
+
+  Ul = replaceinds(U, (rinds..., ur), (linds..., ulnew))
+
+  D_mapped = map_diag(f, D)
+  D_mapped = replaceind(D_mapped, ul, ulnew)
```

Comment:
We could factor out some of this indexing logic into an `eigendecomp` function as suggested here: https://github.com/ITensor/ITensors.jl/issues/341. I.e. if we define `eigendecomp`, you could then just rewrite this as:
```julia
function eigendecomp(a::ITensor, linds, rinds; kwargs...)
  d, vr = eigen(a, linds, rinds; kwargs...)
  # Reindexing logic
  return vr, d, vl
end

function map_eigvals(a::ITensor, inds...; kwargs...)
  vr, d, vl = eigendecomp(a, inds...; kwargs...)
  return vr * map_diag(f, d) * vl
end
```

Reply:
Note that the non-Hermitian case is a bit more involved, as discussed in #341, since you have to compute the left and right eigenvectors with two calls to `eigen` and then normalize them against each other. We could leave that for future work, up to you.

Reply:
Yeah this seems like a good separation of logic to follow

Reply:
I will also add a `#TODO` for the non-Hermitian case